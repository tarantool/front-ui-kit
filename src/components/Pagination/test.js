import React from 'react';
import renderer from 'react-test-renderer';
import { Pagination } from './index';


const changePage = () => {};
const setPageSize = () => {};

describe('Pagination', () => {
  it('render simple', () => {
    const tree = renderer.create(
      <Pagination
        page={2}
        pageSize={10}
        items={300}
        onPageChange={changePage}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with total', () => {
    const tree = renderer.create(
      <Pagination
        showTotal
        page={2}
        pageSize={10}
        items={300}
        onPageChange={changePage}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with page size changer', () => {

    const tree = renderer.create(
      <Pagination
        page={2}
        pageSize={10}
        items={300}
        onPageChange={changePage}
        setPageSize={setPageSize}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with page size changer and total', () => {

    const tree = renderer.create(
      <Pagination
        showTotal
        page={2}
        pageSize={10}
        items={300}
        onPageChange={changePage}
        setPageSize={setPageSize}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
