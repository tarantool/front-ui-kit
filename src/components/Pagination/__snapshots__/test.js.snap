// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination render simple 1`] = `
<div
  className="css-zjik7"
>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1jwstkt"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
  <div>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="css-s5x2y9"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      ...
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      30
    </button>
  </div>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1qr49le"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Pagination renders with page size changer 1`] = `
<div
  className="css-zjik7"
>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1jwstkt"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
  <div>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="css-s5x2y9"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      ...
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      30
    </button>
  </div>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1qr49le"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
  <div
    className="css-17vatmo"
    onClick={[Function]}
  >
    <button
      className="css-1mocbyz"
      type="button"
    >
      10 / page 
      <svg
        className="css-10s1w"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Pagination renders with page size changer and total 1`] = `
<div
  className="css-zjik7"
>
  <div
    className="css-1obq1rs"
  >
    21
    -
    30
     of 
    300
     items
  </div>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1jwstkt"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
  <div>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="css-s5x2y9"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      ...
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      30
    </button>
  </div>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1qr49le"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
  <div
    className="css-17vatmo"
    onClick={[Function]}
  >
    <button
      className="css-1mocbyz"
      type="button"
    >
      10 / page 
      <svg
        className="css-10s1w"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Pagination renders with total 1`] = `
<div
  className="css-zjik7"
>
  <div
    className="css-1obq1rs"
  >
    21
    -
    30
     of 
    300
     items
  </div>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1jwstkt"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
  <div>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="css-s5x2y9"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      ...
    </button>
    <button
      className="css-1x4gift"
      onClick={[Function]}
      type="button"
    >
      30
    </button>
  </div>
  <div>
    <button
      className="css-nbfjew"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        className="css-1qr49le"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`PaginationControlled render simple 1`] = `
<div
  className="css-zjik7"
>
  <button
    className="css-53w1bj"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="css-1jwstkt"
      viewBox="0 0 14 14"
    >
      <use
        xlinkHref="#chevron-up"
      />
    </svg>
  </button>
  <span
    className="css-xqrn7q"
  >
    3
  </span>
  <button
    className="css-53w1bj"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="css-1qr49le"
      viewBox="0 0 14 14"
    >
      <use
        xlinkHref="#chevron-up"
      />
    </svg>
  </button>
</div>
`;

exports[`PaginationControlled renders with page size changer 1`] = `
<div
  className="css-zjik7"
>
  <button
    className="css-53w1bj"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="css-1jwstkt"
      viewBox="0 0 14 14"
    >
      <use
        xlinkHref="#chevron-up"
      />
    </svg>
  </button>
  <span
    className="css-xqrn7q"
  >
    3
  </span>
  <button
    className="css-53w1bj"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    <svg
      className="css-1qr49le"
      viewBox="0 0 14 14"
    >
      <use
        xlinkHref="#chevron-up"
      />
    </svg>
  </button>
  <div
    className="css-17vatmo"
    onClick={[Function]}
  >
    <button
      className="css-b3lbo0"
      type="button"
    >
      10 / page 
      <svg
        className="css-10s1w"
        viewBox="0 0 14 14"
      >
        <use
          xlinkHref="#chevron-up"
        />
      </svg>
    </button>
  </div>
</div>
`;
